include/pmacc/particles/compileTime/FindByNameOrType.hpp

```diff
@@ -0,0 +1,101 @@
+/* Copyright 2018 Rene Widera
+ *
+ * This file is part of PMacc.
+ *
+ * PMacc is free software: you can redistribute it and/or modify
+ * it under the terms of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with PMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "pmacc/traits/GetCTName.hpp"
+#include "pmacc/compileTime/errorHandlerPolicies/ThrowValueNotFound.hpp"
+
+#include <boost/mpl/placeholders.hpp>
+#include <boost/mpl/copy_if.hpp>
+#include <boost/type_traits/is_same.hpp>
+#include <boost/mpl/empty.hpp>
+#include <boost/mpl/or.hpp>
+#include <boost/mpl/front.hpp>
+#include <boost/type_traits/is_same.hpp>
+
+
+namespace pmacc
+{
+namespace particles
+{
+namespace compileTime
+{
+
+    /* find a type within a sequence by name or the type it self
```

Comment:
itself

Reply:
you don't get the type by "a name" but specifically by the `pmacc::traits::GetCTName` trait. maybe we write that in the long description?